api/src/movie/dao/movie.dao.ts

```diff
@@ -63,25 +72,42 @@ export class MovieDao {
       ...(query?.listId && {
         list: {
           some: {
-            id: query?.listId,
+            listId: query?.listId,
           },
         },
       }),
     };
 
     const [movies, count] = await Promise.all([
-      this.prisma.movie.findMany({
-        where: queryCondition,
-        take: query?.per_page || 10,
-        skip: (query?.page_number || 0) * (query?.per_page || 10),
+      this.prisma.listMovie.findMany({
+        where: query?.listId ? { listId: query?.listId } : {},
+        include: {
+          Movie: true,
+        },
+        orderBy: {
+          createdAt: 'desc',
+        },
+        take: query?.per_page || PER_PAGE,
+        skip:
+          (query?.page_number || PAGE_NUMBER) * (query?.per_page || PER_PAGE),
       }),
 
       this.prisma.movie.count({
         where: queryCondition,
       }),
     ]);
 
-    return { movies, count };
+    const moviesWithCreatedAt = movies.map((listMovie) => {
+      const { Movie: movie, createdAt } = listMovie;
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { updatedAt, ...rest } = movie;
+      rest.createdAt = createdAt;
+      return {
+        ...rest,
+      };
+    });
```

Comment:
We can use `select` in `findMany` to select the fields we want, like:

```js
const getSailor: Sailor = await prisma.sailor.findUnique({
  where: {
    id: 22,
  },
  select: {
      name: true,
      boat: true,
    },
})
```

and then reformat the data with a decorator